rhea/system/memmap/memmap.py

```diff
@@ -228,18 +228,18 @@ def map_from_generic(self, generic):
         bus.  This is a module and returns myhdl generators
 
         Arguments
-        ---------
+
         generic: The generic memory-mapped bus, all the memory-mapped
         supported modules use the *generic* bus internally.  This provides
         an agnostic bus interface to all the modules.
 
         Returns
-        -------
+
         myhdl generators
         """
         raise NotImplementedError
 
-    def peripheral_regfile(self, regfile, name, base_address=0):
+    def peripheral_regfile(self, glbl, regfile, name, base_address=0):
```

Comment:
I think the confusion stems from the fact that the method `def peripheral_regfile(self, regfile, name, base_address=0):` in `class MemoryMapped(MemorySpace):` has a different argument list than `def peripheral_regfile(self, glbl, regfile, name, base_address=0):` in the childs objects in avalon, axi4, barebone, and wishbone.

If we're overriding methods, shouldn't the argument lists line up?